### DISCUSSION TITLE
How can I implement a GNN which consider the edge weight in aggregation?

### DISCUSSION THREAD

**Discussion by xin-ran-w** [pyg-team/pytorch_geometric#5609] [Q&A]
Hello, I have an idea to implement a GNN, which predicts node-level features (each node feature's dimension is 1).

Assuming that, there is a vertice $v$ (orange node in the figure) and its neighbours $\mathcal{N}(v)$ (blue nodes in the figure).
There are two relation types between $v$ and its neighbours. 
The orange line whose value is $0.6$ represents the improve relation. It will increase the value of vertice $v$ through the message passing.
The blue line whose value is $-0.4$ represents the impress relation. It will decrease the value of vertice $v$ through the message passing.

I do not know which GNN I should use. 

<img width="181" alt="image" src="https://user-images.githubusercontent.com/87413406/193975996-2acd8deb-d684-4deb-8e72-03ffd4e67697.png">

**Comment by rusty1s** [comment]
I suggest using `GraphConv` for this, which supports both positive and negative edge weights.

**Reply by xin-ran-w** [reply]
Thank you, I have tried GCN, however, I do not know how to apply edge weight in GCN.

**Reply by rusty1s** [reply]
```python
conv = GraphConv(16, 32)
conv(x, edge_index, edge_weight)
```
where `edge_weight` is a vector holding the weight for each edge.

**Reply by xin-ran-w** [reply]
Thank you, I'll try! If it's effective, I will mark this as an answer.
